hail/python/hailtop/batch/backend.py

```diff
@@ -199,25 +199,30 @@ def symlink_input_resource_group(r):
             resource_defs = [r._declare(tmpdir) for r in job._mentioned]
             env = [f'export {k}={v}' for k, v in job._env.items()]
 
+            job_shell = job._shell if job._shell else self.DEFAULT_SHELL
+
+            defs = '; '.join(resource_defs) + '; ' if resource_defs else ''
+            joined_env = '; '.join(env) + '; ' if env else ''
+            cmd = " && ".join(f'{{\n{x}\n}}' for x in job._command)
+
+            job_script = shq(joined_env + defs + cmd)
+
             if job._image:
-                defs = '; '.join(resource_defs) + '; ' if resource_defs else ''
-                joined_env = '; '.join(env) + '; ' if env else ''
-                cmd = " && ".join(f'{{\n{x}\n}}' for x in job._command)
+
                 memory = f'-m {job._memory}' if job._memory else ''
                 cpu = f'--cpus={job._cpu}' if job._cpu else ''
 
                 lines.append(f"docker run "
+                             "--entrypoint=''"
                              f"{self._extra_docker_run_flags} "
                              f"-v {tmpdir}:{tmpdir} "
                              f"-w {tmpdir} "
                              f"{memory} "
                              f"{cpu} "
-                             f"{job._image} /bin/bash "
-                             f"-c {shq(joined_env + defs + cmd)}")
+                             f"{job._image} "
+                             f"{job_shell} -c {job_script}")
             else:
-                lines += env
-                lines += resource_defs
-                lines += job._command
+                lines.append(f"{job_shell} -c {job_script}")
```

Comment:
Can you put the shq here, or call this quoted_job_script?  Quoting is so important, it should be clear it is happening at the point of use, otherwise this line looks wrong.